daemon/cluster/cluster.go

```diff
@@ -1245,7 +1370,12 @@ func (c *Cluster) RemoveNetwork(input string) error {
 }
 
 func (c *Cluster) populateNetworkID(ctx context.Context, client swarmapi.ControlClient, s *types.ServiceSpec) error {
-	for i, n := range s.Networks {
+	networks := s.TaskTemplate.Networks
+	if len(networks) == 0 {
+		networks = s.Networks
```

Comment:
Yeah, this is one of the places we need to change as per fallback logic. Will add a note.